src/cargo-kani/src/call_single_file.rs

```diff
@@ -0,0 +1,99 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use anyhow::Result;
+use std::ffi::OsString;
+use std::path::{Path, PathBuf};
+use std::process::Command;
+
+use crate::session::KaniSession;
+use crate::util::alter_extension;
+
+impl KaniSession {
+    /// Used by `kani` and not `cargo-kani` to process a single Rust file into a `.symtab.json`
+    pub fn compile_single_rust_file(&self, file: &Path) -> Result<PathBuf> {
+        let output_filename = alter_extension(file, "symtab.json");
+
+        {
+            let mut temps = self.temporaries.borrow_mut();
+            temps.push(output_filename.clone());
+            temps.push(alter_extension(file, "type_map.json"));
+            temps.push(alter_extension(file, "kani-metadata.json"));
```

Comment:
Can we create a little utility class to generate these extensions for us? Another alternative is to create a constant for each type of extension. I noticed that they are hard-coded in different places.

Reply:
While they're hardcoded, they each appear exactly once (except "restrictions.json" which is a design flaw I might try to correct now...)

Reply:
Ok. Maybe once you merge this I can add a little utility module that can be used by kani / cargo-kani and the kani-compiler.